types/src/error.rs

```diff
@@ -265,30 +256,6 @@ impl serde::Serialize for ErrorCode {
 	}
 }
 
-/// Error that occurs when a call failed.
-#[derive(Debug, thiserror::Error)]
-pub enum CallError {
-	/// Invalid params in the call.
-	#[error("Invalid params in the call: {0}")]
-	InvalidParams(#[source] anyhow::Error),
-	/// The call failed (let jsonrpsee assign default error code and error message).
-	#[error("RPC call failed: {0}")]
-	Failed(#[from] anyhow::Error),
-	/// Custom error with specific JSON-RPC error code, message and data.
-	#[error("RPC call failed: {0:?}")]
-	Custom(ErrorObjectOwned),
-}
-
-impl CallError {
-	/// Create `CallError` from a generic error.
-	pub fn from_std_error<E>(err: E) -> Self
-	where
-		E: std::error::Error + Send + Sync + 'static,
-	{
-		CallError::Failed(err.into())
-	}
-}
```

Comment:
Ah ok, so previously user errors were being turned into thie `Failed` variant, and now the user describes how to turn them into an `ErrorObjectOwned` basically? This makes a lot of sense to me; give users the ability to set the error code etc :)

Reply:
Yeah, this was a way to easily convert any other StdError into a `ErrorObject` and let's jsonrpsee pick the error code.

This was a bad idea it's better that folks take care of the error code etc and explicitly.